Moonray docs

evilnick opened the pull request:
- adds moonray sphinx instance for generating docs
- adds substitution terms
- converts existing docs to use substitution
- fixes a few CAPI docs

Currently the Moonray docs are available here - https://canonical-moonray.readthedocs-hosted.com/docs-moonray-staging/
N.B. there is minimal content there yet

NB this replaces #578

evilnick commented:
Also adds a workflow to integrate standard doc test - I have left this as run on demand at the moment because it is going to spew out millions of errors

evilnick commented:
You can check no Canonical kubernetes was hurt during this operation by viewing [these docs](https://canonical-ubuntu-documentation-library--581.com.readthedocs.build/canonical-kubernetes/581/capi/)
The new, largely empty Moonray stuff is [here](https://canonical-moonray--581.com.readthedocs.build/581/)

louiseschmidtgen commented on docs/moonray/explanation/about.md:
```diff
@@ -0,0 +1,7 @@
+# What is {{product}}?
+
+At its core, {{product}} is a full implementation of upstream
+[Kubernetes] delivered in a compact, secure, reliable [snap] package. As the
```
Could you please add the links for Kubernetes and the snap?

louiseschmidtgen commented on docs/moonray/howto/install.md:
````diff
@@ -0,0 +1,3 @@
+```{include} ../../src/snap/howto/install/snap.md
+
````
<img width="1197" alt="image" src="https://github.com/user-attachments/assets/ed2d1ece-de0a-4788-892f-2302c9bf0bb6">
The note's Installing leads us to the how-to page, there's probably a page missing could you please add a dummy missing page.

louiseschmidtgen replied:
Same goes for the other links on this page. Do we need to bring in the channel page and bootstrap ref guide?

evilnick replied:
I think we will either need to put conditional blocks in there or harmonise the directory structure first. I have just added a todo note for now - bringing in those pages will bring in more unresolvable links right now

louiseschmidtgen commented on docs/moonray/index.md:
```diff
@@ -0,0 +1,76 @@
+<!-- LINKS -->
+
+[Code of Conduct]: https://ubuntu.com/community/ethos/code-of-conduct
+[contribute]: index
```
Let's add a TODO here or add the contribution page.

louiseschmidtgen commented on docs/moonray/index.md:
```diff
@@ -0,0 +1,76 @@
+
+[Code of Conduct]: https://ubuntu.com/community/ethos/code-of-conduct
+[contribute]: index
+[overview page]: /explanation/about
```
This doesn't render. Let's add a TODO here or add the contribution page please.

evilnick replied:
I'm just going to cut the bits we haven't added content for at this point

louiseschmidtgen reviewed (commented):
This is looking great, please add some TODOs or add missing files so that we don't accidentally forget about broken links.
Please check the the mirrored pages like `../../src/snap/howto/install/snap.md` and their links.

evilnick commented:
thanks @louiseschmidtgen. I added a TODO on one file and just removed a lot of references elsewhere to resolve the broken links

Some of how this is going to work moving forward is going to require some careful editing of the 'common' pages and probably some changes to the structure so that things just work, but thats for another PR

bschimke95 reviewed (commented):
great work @evilnick 

An initial question:

What is the reason for the moonray docs to be in `docs/moonray` and duplicate the setup like this? Looking at the code, it seems a bit odd to have `docs/src/...` and `docs/moonray`. I'd either move `moonray` to the `src` folder or rename `src` to something that reflects that these are the default docs (e.g. `core`?).

evilnick commented on .github/workflows/automatic-doc-checks.yml:
```diff
@@ -0,0 +1,14 @@
+  documentation-checks:
+    uses: canonical/documentation-workflows/.github/workflows/documentation-checks.yaml@main
+    with:
+      working-directory: 'docs/moonray'
```
:+1:

bschimke95 commented:
Thanks for the context @evilnick 
The envisioned structure makes a lot of sense and also splitting up that work into multiple PRs is very reasonable.

bschimke95 reviewed (approved):
LGTM
